bciers/apps/reporting/src/data/jsonSchema/signOff/signOff.ts

```diff
@@ -4,54 +4,107 @@ import {
+    default: false,
+  },
+  acknowledgement_of_records: {
+    title:
+      "I understand that the Ministry responsible for the administration and enforcement of the Greenhouse Gas Industrial Reporting and Control Act may require records from the Operator evidencing the truth of this report.",
+    type: "boolean",
+    default: false,
+  },
+  signature: {
+    type: "string",
+    format: "signature",
+    title: "Please add your signature by typing your name here:",
+  },
+  date: {
+    type: "string",
+    title: "Date signed",
+  },
+  ...(flow === ReportingFlow.EIO
+    ? {
+        acknowledgement_of_errors: {
+          title:
+            "I understand that any errors, omissions, or misstatements provided in this report can lead to administrative penalties.",
+          type: "boolean",
+          default: false,
+        },
+      }
+    : {
+        acknowledgement_of_information: {
+          title:
+            "I understand that this information is being collected for the purpose of emission reporting under the Greenhouse Gas Industrial Reporting and Control Act and may be disclosed to the Ministry responsible for the administration and enforcement of the Carbon Tax Act.",
+          type: "boolean",
+          default: false,
+        },
+      }),
+});
+
+const getSupplementaryFields = (
+  isRegulated: boolean,
+): RJSFSchema["properties"] => ({
+  editing_note: {
+    title:
+      "By editing the original submission, please confirm that you understand the following:",
+    type: "string",
+  },
+  acknowledgement_of_new_version: {
+    title:
+      "I understand that, by submitting these changes, I am creating a new version of this annual report that will, effective immediately, be the annual report for the reporting and/or compliance period that it pertains to.",
+    type: "boolean",
+    default: false,
+  },
+  ...(isRegulated && {
+    acknowledgement_of_corrections: {
       title:
-        "I understand that this information is being collected for the purpose of emission reporting under the Greenhouse Gas Industrial Reporting and Control Act and may be disclosed to the Ministry responsible for the administration and enforcement of the Carbon Tax Act.",
+        "I understand that the correction of any errors, omissions, or misstatements in the new submission of this report may lead to an additional compliance obligation, and, if submitted after the compliance obligation deadline, applicable interest.",
       type: "boolean",
       default: false,
     },
-    acknowledgement_of_possible_costs: {
+  }),
+});
+
+export const buildSignOffSchema = (
+  isSupplementary: boolean,
+  isRegulated: boolean,
+  flow: string,
+): RJSFSchema => {
+  const properties: RJSFSchema["properties"] = {
+    ...baseFields(flow),
+  };
+
+  if (isSupplementary) {
+    properties.supplementary = {
+      type: "object",
+      properties: getSupplementaryFields(isRegulated),
+    };
+  } else if (flow !== ReportingFlow.EIO) {
```

Comment:
This is a lot of if/then/else that don't follow the same patterns, nor are tested.
Mixing conditions about supplementary, flows etc. is a recipe for multiplying cases having to handle and test a lot of possible options, or introducing bugs when we forget about certain cases.

We can talk about it if you want, we propably want to expose mutliple json schemas/pages through our factories instead